sale_margin_delivered/views/sale_margin_delivered_view.xml

```diff
@@ -5,10 +5,6 @@
     <record id="view_order_form" model="ir.ui.view">
         <field name="model">sale.order</field>
         <field name="inherit_id" ref="sale_margin.sale_margin_sale_order_line_form" />
-        <field
-            name="groups_id"
-            eval="[(4,ref('sale_margin_security.group_sale_margin_security'))]"
-        />
```

Comment:
You have to add the groups to fields themselves. Not just removing them from the view.